packages/contracts-bedrock/contracts/universal/ProxyAdmin.sol

```diff
@@ -182,72 +186,68 @@ contract ProxyAdmin is Owned {
+            addressManager.transferOwnership(_newAdmin);
         } else {
             revert("ProxyAdmin: unknown proxy type");
         }
     }
 
     /**
-     * @dev Upgrades `proxy` to `implementation`. This contract must be the admin of `proxy`.
+     * @notice Changes a proxy's implementation contract.
      *
-     * @param proxy          The address of the proxy.
-     * @param implementation The address of the implementation.
+     * @param _proxy          Address of the proxy to upgrade.
+     * @param _implementation Address of the new implementation address.
      */
-    function upgrade(Proxy proxy, address implementation) public onlyOwner {
-        ProxyType proxyType = proxyType[address(proxy)];
-
+    function upgrade(address _proxy, address _implementation) public onlyOwner {
+        ProxyType proxyType = proxyType[_proxy];
         if (proxyType == ProxyType.ERC1967) {
-            proxy.upgradeTo(implementation);
+            Proxy(payable(_proxy)).upgradeTo(_implementation);
         } else if (proxyType == ProxyType.Chugsplash) {
-            L1ChugSplashProxy(payable(proxy)).setStorage(
+            L1ChugSplashProxy(payable(_proxy)).setStorage(
                 0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc,
-                bytes32(uint256(uint160(implementation)))
+                bytes32(uint256(uint160(_implementation)))
             );
         } else if (proxyType == ProxyType.ResolvedDelegate) {
-            string memory name = implementationName[address(proxy)];
-            addressManager.setAddress(name, implementation);
+            string memory name = implementationName[_proxy];
+            addressManager.setAddress(name, _implementation);
         } else {
             revert("ProxyAdmin: unknown proxy type");
         }
     }
 
     /**
-     * @dev Upgrades `proxy` to `implementation` and calls a function on the new implementation.
-     *      This contract must be the admin of `proxy`.
+     * @notice Changes a proxy's implementation contract and delegatecalls the new implementation
+     *         with some given data. Useful for atomic upgrade-and-initialize calls.
      *
-     * @param proxy           The proxy to call.
-     * @param implementation  The implementation to upgrade the proxy to.
-     * @param data            The calldata to pass to the implementation.
+     * @param _proxy          Address of the proxy to upgrade.
+     * @param _implementation Address of the new implementation address.
+     * @param _data           Data to trigger the new implementation with.
      */
     function upgradeAndCall(
-        Proxy proxy,
-        address implementation,
-        bytes memory data
+        address _proxy,
+        address _implementation,
+        bytes memory _data
     ) external payable onlyOwner {
-        ProxyType proxyType = proxyType[address(proxy)];
-
+        ProxyType proxyType = proxyType[_proxy];
```

Comment:
There is a compiler warning where where the `proxyType` variable overshadows the `proxyType` mapping. Perhaps we rename the variable to `ptype` or `typ`

Reply:
Resolved